### DISCUSSION TITLE
Shouldn't the result type for find queries be like <ModelBeingSearched> | null

### DISCUSSION THREAD

**Discussion by FeMaffezzolli** [prisma/prisma#13378] [Q&A] [answered]
```typescript
const user = await this.prisma.user.findUnique({
  where: {
    email,
  },
});
```
The type type for the user const is `User`. But it feels like it shouldn't be. It should be `User | null`.

Is there a way to fix that? What you guys think about?

**Comment by nurul3101** [comment]
Hey @FeMaffezzolli 👋 
It's showing as User | null for me. Could you share your types or a simple reproduction?
On a sidenote, you might be interested in having a look at this guide: [Operating against partial structures of model types](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types).

<img width="554" alt="Screenshot 2022-06-10 at 4 46 32 PM" src="https://user-images.githubusercontent.com/13049676/173053660-507985c1-176d-471f-b05c-4c569ee51a8a.png">

**Reply by FeMaffezzolli** [reply]
Hi @nurul3101 , thanks for your answer 👋🏻.

I did not mention before, but I'm prisma along with NestJs.

**Here is my schema for user**

![image](https://user-images.githubusercontent.com/26548010/173239154-3328ba6b-75fb-43e0-bc98-6d6a3316f11f.png)

**Here is my type file for the findFirst method on user:**

```typescript

    /**
     * Find the first User that matches the filter.
     * Note, that providing `undefined` is treated as the value not being there.
     * Read more here: https://pris.ly/d/null-undefined
     * @param {UserFindFirstArgs} args - Arguments to find a User
     * @example
     * // Get one User
     * const user = await prisma.user.findFirst({
     *   where: {
     *     // ... provide filter here
     *   }
     * })
    **/
    findFirst<T extends UserFindFirstArgs,  LocalRejectSettings = T["rejectOnNotFound"] extends RejectOnNotFound ? T['rejectOnNotFound'] : undefined>(
      args?: SelectSubset<T, UserFindFirstArgs>
    ): HasReject<GlobalRejectSettings, LocalRejectSettings, 'findFirst', 'User'> extends True ? CheckSelect<T, Prisma__UserClient<User>, Prisma__UserClient<UserGetPayload<T>>> : CheckSelect<T, Prisma__UserClient<User | null >, Prisma__UserClient<UserGetPayload<T> | null >>

```

**Here you can see that the type is still incorrect.**

![image](https://user-images.githubusercontent.com/26548010/173239039-9251d1eb-5b89-4e05-adc3-23918d36d3a0.png)

**Here are my packages versions:**

```json
"dependencies": {
    "@googlemaps/google-maps-services-js": "^3.3.14",
    "@nestjs-modules/mailer": "1.5.1",
    "@nestjs/axios": "^0.0.7",
    "@nestjs/common": "^8.0.0",
    "@nestjs/config": "^2.0.0",
    "@nestjs/core": "^8.0.0",
    "@nestjs/jwt": "^8.0.0",
    "@nestjs/passport": "^8.2.1",
    "@nestjs/platform-express": "^8.0.0",
    "@nestjs/swagger": "^5.2.1",
    "@nestjs/terminus": "^8.0.6",
--> "@prisma/client": "^3.15.1",
    "argon2": "^0.28.5",
    "aws-sdk": "^2.1145.0",
    "class-transformer": "^0.5.1",
    "class-validator": "^0.13.2",
    "handlebars": "^4.7.7",
    "hbs": "^4.2.0",
    "joi": "^17.6.0",
    "lodash": "^4.17.21",
    "luxon": "^2.4.0",
    "morgan": "^1.10.0",
    "nodemailer": "^6.7.5",
    "passport": "^0.5.2",
    "passport-jwt": "^4.0.0",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^7.2.0",
    "stripe": "^9.6.0",
    "swagger-ui-express": "^4.3.0",
    "uuid": "^8.3.2"
  },
  "devDependencies": {
    "@faker-js/faker": "^6.3.1",
    "@nestjs/cli": "^8.0.0",
    "@nestjs/schematics": "^8.0.0",
    "@nestjs/testing": "^8.0.0",
    "@types/chai": "^4.3.1",
    "@types/express": "^4.17.13",
    "@types/jest": "27.4.1",
    "@types/lodash": "^4.14.182",
    "@types/luxon": "^2.3.2",
    "@types/morgan": "^1.9.3",
    "@types/multer": "^1.4.7",
    "@types/node": "^17.0.33",
    "@types/nodemailer": "^6.4.4",
    "@types/passport-jwt": "^3.0.6",
    "@types/supertest": "^2.0.11",
    "@types/uuid": "^8.3.4",
    "@typescript-eslint/eslint-plugin": "^5.0.0",
    "@typescript-eslint/parser": "^5.0.0",
    "chai": "^4.3.6",
    "dotenv-cli": "^5.1.0",
    "eslint": "^8.0.1",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-prettier": "^4.0.0",
    "jest": "^27.2.5",
    "pactum": "^3.1.7",
    "pactum-matchers": "^1.1.1",
    "prettier": "^2.3.2",
--> "prisma": "^3.15.1",
    "prisma-dbml-generator": "^0.9.1",
    "source-map-support": "^0.5.20",
    "supertest": "^6.1.3",
    "ts-jest": "^27.0.3",
    "ts-loader": "^9.2.3",
    "ts-node": "^10.7.0",
    "tsconfig-paths": "^3.10.1",
    "typescript": "4.7.2"
  },
```

And finally here is my PrismaService class

![image](https://user-images.githubusercontent.com/26548010/173239679-191996b8-d3c1-4d9b-b9f8-3dc7c2fde4bf.png)

**Comment by aqrln** [comment, marked as answer]
Your `tsconfig.json` doesn't have `"strict": true` or at least `"strictNullChecks": true` set in `"compilerOptions"`.

Without `strictNullChecks`, there's no such type as `User | null` in TypeScript, it merges it to just `User`.

**Reply by FeMaffezzolli** [reply]
Thanks @aqrln , you are correct.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
